src/unsolicited_responses.rs

```diff
@@ -0,0 +1,37 @@
+use enumset::EnumSet;
+use std::sync::mpsc;
+
+use super::types::{UnsolicitedResponse, UnsolicitedResponseCategory};
+
+#[derive(Debug, Clone)]
+pub struct UnsolicitedResponseSender {
+    sender: mpsc::Sender<UnsolicitedResponse>,
+    allow: EnumSet<UnsolicitedResponseCategory>,
+}
+
+impl UnsolicitedResponseSender {
+    pub(crate) fn new(sender: mpsc::Sender<UnsolicitedResponse>) -> UnsolicitedResponseSender {
+        UnsolicitedResponseSender {
+            sender,
+            allow: EnumSet::empty(),
+        }
+    }
+
+    // Set the new filter mask, and remove unwanted responses from the current queue.
+    pub(crate) fn request(
+        &mut self,
+        receiver: &mpsc::Receiver<UnsolicitedResponse>,
+        mask: EnumSet<UnsolicitedResponseCategory>,
+    ) {
+        self.allow = mask;
+        for message in receiver.try_iter() {
```

Comment:
This also seems pretty strange -- what about future messages? Are they never forwarded then?

Reply:
Long time no see :) Did you ever get around to doing this rewrite?